File: Problems/0100_0199/0133_Clone_Graph/Project_Go/solution/OperateNode.go
package solution

import (
	"strings"
)

func CreateTreeNode(flds string) *Node {
	fldsArr := strings.Split(flds, "],[")
	data := make([][]int, len(fldsArr))
	for i := 0; i < len(data); i++ {
		data[i] = StringToIntArray(fldsArr[i])
	}

	nodes := make([]*Node, len(data))
	for i, _ := range data {
		nodes[i] = &Node{i + 1, nil}
	}
	for i, _ := range nodes {
		tempNodes := make([]*Node, 0)
		for j, _ := range data[i] {
			tempNodes = append(tempNodes, nodes[data[i][j]-1])
		}
		nodes[i].Neighbors = tempNodes
	}
	return nodes[0]
}

func NodeToString(node *Node) string {
	if node == nil {
		return "[]"
	} else if node.Neighbors == nil {
		return "[[]]"
	}
	resultStr := ""
	data := make(map[int][]int, 0)
	for _, nei := range node.Neighbors {
		data[node.Val] = append(data[node.Val], nei.Val)
	}

	for _, nei := range node.Neighbors {
		_, exists := data[nei.Val]
		if !exists {
			for _, nei2 := range nei.Neighbors {
				data[nei.Val] = append(data[nei.Val], nei2.Val)
			}
		}

		for _, nei2 := range nei.Neighbors {
			_, exists := data[nei2.Val]
			if !exists {
				for _, nei3 := range nei2.Neighbors {
					data[nei2.Val] = append(data[nei2.Val], nei3.Val)
				}
			}
		}
	}
	resultStr = ""
	for k, _ := range data {
		if resultStr == "" {
			resultStr += "[" + IntArrayToString(data[k]) + "]"
		} else {
			resultStr += ",[" + IntArrayToString(data[k]) + "]"
		}
	}
	return resultStr
}

File: Problems/0100_0199/0133_Clone_Graph/Project_Python3/Node/OperateNode.py
from typing import List, Dict, Tuple
from .Node import Node

class OperateNode:
    def createNode(self, data: List[int]) -> Node:
        nodes = [None for n in range(len(data))]
        for i, _ in enumerate(data):
            nodes[i] = Node(i + 1, None)
        for i, _ in enumerate(nodes):
            nodes[i].neighbors = [nodes[col - 1] for col in data[i]]
        return nodes[0]

    def nodeToString(self, node: Node) -> str:
        if not node:
            return "[]"
        elif node.neighbors is None:
            return "[[]]"
        resultStr = ""
        data = {}
        data[node.val] = [nei.val for nei in node.neighbors]
        for nei in node.neighbors:
            data[nei.val] = [nei2.val for nei2 in nei.neighbors]
            for nei2 in nei.neighbors:
                if nei2.val not in data:
                    data[nei2.val] = [nei3.val for nei3 in nei2.neighbors]
        resultStr = ""
        for k in data.keys():
            if resultStr == "":
                resultStr += str(data[k])
            else:
                resultStr += "," + str(data[k])
        return resultStr
